File: java/problem66.java
class Solution {
    public int[] plusOne(int[] digits) {
        int size=digits.length;
        for(int i=size-1;i>=0;i--){
            if(digits[i]!=9){
                digits[i]++;
                break;   
            }else{
                digits[i]=0;
            }    
        }
        if(digits[0]==0){
            int[] result=new int[size+1];
            result[0]=1;
            for(int i=1;i<result.length;i++){
                result[i]=0;
            }
            return result;
        }else{
            return digits;
        }
    }
}

File: python/problem66.py
class Solution:
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        x=''.join(str(i) for i in digits)
        k=int(x)+1
        li=[]
        for i in str(k):
            li.append(int(i))
        return li
